libs/model/src/middleware/guards.ts

```diff
@@ -106,3 +107,53 @@ export function mustBeAuthorizedComment(actor: Actor, auth?: AuthContext) {
     comment_id: number;
   };
 }
+
+/**
+ * Guards for starting and ending dates to be in a valid date range
+ * @param start_date start date
+ * @param end_date end date
+ * @param startDaysInTheFuture number of days in the future that the start date must be
+ * @param minDaysInRange
+ * @returns { start, end } start and end dates as moment objects
+ */
+export function mustBeValidDateRange(
+  start_date: Date,
+  end_date: Date,
+  startDaysInTheFuture = 1,
+  minDaysInRange = 1,
+) {
+  const today = moment(new Date());
```

Comment:
We should probably look at moving away from moment.js given that it is self-declared legacy and in-maintenance mode.
Something like Day.js or date-fns.

Reply:
100% - we need a maintenance ticket here https://github.com/hicommonwealth/commonwealth/issues/5400